### PR TITLE
Marketplace Module

### PR DESCRIPTION
**Pull request by ayshptk**

### CODE REVIEW

File: thirdweb/sdk.py [hunk 1]
```diff
@@ -120,18 +121,28 @@ def get_pack_module(self, address: str) -> PackModule:
         return self.__pack_module
 
     @set_default_account
-    def get_market_module(self, address: str) -> MarketModule:
+    def get_market_module(self, address: str, old: bool = True) -> MarketModule:
```
Comment (atbe) [inline thirdweb/sdk.py:124]: lets rename `vold` to `isV1`

Comment (atbe) [inline thirdweb/sdk.py:124]: Mark as deprecated

**Comment by atbe** [comment]
Be sure to mark this class as `deprecated` https://github.com/nftlabs/nftlabs-sdk-python/blob/master/thirdweb/modules/market.py#L22

File: thirdweb/modules/marketplace.py [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+            raise WrongListingTypeException(
+                listing_id, "Auction", "Direct")
+
+        return self.map_direct_listing(listing)
+
+    def map_direct_listing(self, listing):
+        nftcontract = NftModule(listing[3], self.get_client())
+        return DirectListing(listing[0], assetContractAddress=listing[2], tokenId=listing[3], asset=nftcontract.get(listing[3]), startTimeInSeconds=listing[4], secondsUntilEnd=listing[5], quantity=listing[6], currencyContractAddress=listing[7], buyoutCurrencyValuePerToken=CurrencyModule(listing[7], self.get_client).get_value(listing[8]), buyoutPrice=listing[9], sellerAddress=listing[1], listing_type=[11])
+        
+    def _handle_token_approval(self, asset: str, token_id: int, from_address: str):
```
Comment (joaquim-verges) [inline thirdweb/modules/marketplace.py:189]: this and `map_direct_listing` should be private I believe? same for a bunch of the internal functions below

File: thirdweb/sdk.py [hunk 3]
```diff
@@ -127,18 +128,29 @@ def get_pack_module(self, address: str) -> PackModule:
         return self.__pack_module
 
     @set_default_account
-    def get_market_module(self, address: str) -> MarketModule:
+    def get_market_module(self, address: str, isv1: bool = True) -> MarketModule:
```
Comment (joaquim-verges) [inline thirdweb/sdk.py:131]: the deprecation message is wrong here - should direct ppl to use `get_marketplace_module`

File: thirdweb/modules/marketplace.py [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+    """
+    address: str
+    """
+    Address of the module
+    """
+    """
+    Address of the market contract.
+    """
+    __abi_module: Marketplace
+    MAX_BPS: 1000
```
Comment (joaquim-verges) [inline thirdweb/modules/marketplace.py:37]: missing a 0 here I think - max bps should be 10000 (ten thousand)

File: thirdweb/modules/marketplace.py [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+            listing.currencyContractAddress,
+            listing.reservePricePerToken,
+            listing.buyoutPricePerToken,
+            1,
+        ))
+        receipt = self.execute_tx(tx)
+        return receipt["logs"][0]["args"]["listingId"]
+
+    def make_direct_listing_offer(self, offer: NewOffer):
+        if (offer.currency_contract_address == ZeroAddress or offer.currency_contract_address == "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee"):
```
Comment (joaquim-verges) [inline thirdweb/modules/marketplace.py:103]: we have a constant for native address right?

File: thirdweb/modules/marketplace.py [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+            listing = self.get_listing(listing_id)
+        except:
+            return None
+        if listing.type == 1:
+            return listing
+        valid = self.__is_still_valid_direct_listing(listing)
+        if not valid:
+            return None
+        return listing
+        i += 1
```
Comment (joaquim-verges) [inline thirdweb/modules/marketplace.py:392]: where is this `i` coming from xD

File: thirdweb/sdk.py [hunk 7]
```diff
@@ -127,18 +128,29 @@ def get_pack_module(self, address: str) -> PackModule:
...
         self.__init_module(module)
         self.__market_module = module
         return self.__market_module
 
+    @set_default_account
+    def get_marketplace_module(self, address: str) -> MarketplaceModule:
+        """
+        Returns an instance of the marketplace module
+        """
+        return self.get_market_module(address, old=False)
```
Comment (joaquim-verges) [inline thirdweb/sdk.py:152]: param name is wrong?

**Review by joaquim-verges — COMMENTED** [review]
can we make a lil sanity check script that creates a listing and buys it on mumbai?

**Pull request closed without merging**